src/server.js

```diff
@@ -375,7 +389,7 @@ module.exports = function (classes) {
                 // Try to notify client about failure to authenticate
                 if (Endpoint.hasId(decoded)) {
                   conn.sendReply('Error: Unauthorized', null, decoded.id);
-                  reject(new Error.InvalidParams(UNAUTHORIZED))
+                  return reject(new Error.InvalidParams(UNAUTHORIZED));
```

Comment:
I stumbled upon an article on rising stack today: https://blog.risingstack.com/node-js-best-practices/

```Return on callbacks

One of the problems that still exists in the above example, is that if an Error occurs, then the execution will not stop in the if statement, but will continue. This can lead to lots of unexpected things. As of a rule of thumb, always return on callbacks.

// this example is **STILL BROKEN**, we are fixing it!
function readJSON(filePath, callback) {  
  fs.readFile(filePath, function(err, data) {
    if (err) {
      return callback(err);
    }

    return callback(null, JSON.parse(data));
  });
}```

They say to always return on callbacks.. Since the resolve and reject are callbacks (??), to prevent further execution of code, we call the callback with a return statement.